services/user/Supervisor/ui/src/oauth/db.ts

```diff
@@ -0,0 +1,65 @@
+import { Result } from '../hostInterface';
+import { RecoverableErrorPayload } from '../plugin/errors';
+import { OAUTH_REQUEST_KEY, OAUTH_REQUEST_EXPIRATION } from '../constants';
+import { getSupervisor } from "@psibase/common-lib";
+
+export const supervisor = getSupervisor();
+export interface OauthRequest {
+    id: string,
+    subdomain: string,
+    subpath: string,
+    expiry_timestamp: number,
+}
+
+export const isTypeOauthRequest = (
+    oauthReqRes: OauthRequest | RecoverableErrorPayload,
+): oauthReqRes is OauthRequest => {
+    return (
+        "subdomain" in oauthReqRes &&
+        "subpath" in oauthReqRes &&
+        "expiry_timestamp" in oauthReqRes
+    );
+};
+
+export class ActiveOauthRequest {
+    static async get(id: string): Promise<Result<OauthRequest, RecoverableErrorPayload>> {
+        const oauthReqBytes = await supervisor.functionCall({
+            service: "clientdata",
+            intf: "keyvalue",
+            method: "get",
+            params: [OAUTH_REQUEST_KEY]});
+
+        if (!oauthReqBytes) {
+            throw new Error("No active oauth request found");
+        }
+
+        const oauthReq = JSON.parse(new TextDecoder().decode(oauthReqBytes));
+        if (id != oauthReq.id) {
+            await this.delete();
+            throw new Error("Oauth request id mismatch");
+        }
+
+        let isExpired = (new Date().getUTCSeconds()
```

Comment:
Are you sure you want [`getUTCSeconds`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCSeconds) ? I think you want [`getTime`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime) instead which will return milliseconds, can do `Math.floor(new Date().getTime() / 1000)` for seconds. `getUTCSeconds` is only going to return `0-59`

Reply:
Great catch. Thanks. It hit my brain like seconds-since-the-epoch, and I didn't catch that.